File: math/461.hamming-distance.go
/*
 * @lc app=leetcode id=461 lang=golang
 *
 * [461] Hamming Distance
 */

// @lc code=start
func hammingDistance(x int, y int) int {
	res := 0
	a := x^y
	for a!=0{
		if a%2==1{
			res+=1
		}
		a = a>>1
	}
	return res
}
// @lc code=end

File: math/461.hamming-distance.py
#
# @lc app=leetcode id=461 lang=python3
#
# [461] Hamming Distance
#

# @lc code=start
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        res = 0
        a = x ^ y
        while a:
            if a % 2 == 1:
                res += 1
            a = a >> 1
        return res


# @lc code=end

if __name__ == '__main__':
    s = Solution()
    print(s.hammingDistance(1, 4))
